wordpress-mobile/WordPress-Android
# [Update Detekt Version] Update Detekt to 1.21.0

Parent: #16892
Closes: #16898
Closes: #16899

This PR:
- Updates Detekt to `1.21.0` (from `1.19.0` via `1.20.0`) (f9803cb85794c96be310bf620f8b33a45b4a544a + bb316e90db785a36a6bf08f5726e9dfd87a255b3).
- Updates Detekt configuration to adhere with the update (6462ecc9f8a1d86808d55ae6b0de3af585fc1d09 + 0b8225936a17bff1bcb7ae072a9ab3260a14d2f0)
- 1.20.0 Related:
    - Resolves `UnnecessaryAbstractClass` warnings (d702ca924b2c56f9c5bc24e8768aaff335b9537c)
    - Resolves `TopLevelPropertyNaming` warnings (4ef2603db97bce726d32fff243c6ba9e72f4770d)
- 1.21.0 Related:
    - Suppresses `UseCheckOrError` warnings (bd4df02300cdfe189fac66141c713fb4ed2e2306)
    - Resolves `ExplicitItLambdaParameter` warnings (3046bb71c7375313b0cdf72d0e6927651867304a)
    - Resolves `UseRequire` warnings (0983a08d2a3d19f5bb56055676dfdca5d8596ca5)
    - Suppresses `DestructuringDeclarationWithTooManyEntries` warnings (e6346527ad5d9796d112094fc3d1ff392bd6c666)
    - Suppresses `ReturnCount` warnings (74baf6ee1ff616c15fa08b7f8d250e80d290614a)
    - Resolves `FunctionNaming` warnings (8ba62d7be11f3f1e29106acf996e2c112233462d)
    - Suppresses `ComplexMethod` warning (bd067b69cbae70d980c50e154dc2b2fb5cdb4005)

PS: I recommend reviewing this PR commit-by-commit.

-----

To test:

- Verify that all the CI checks are successful (especially the detekt check).
- Smoke test the app and verify everything works as expected.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
